proxy/coco/src/source.rs

```diff
@@ -513,8 +513,31 @@ pub struct LocalState {
 /// Will return [`Error`] if the repository doesn't exist.
 pub fn local_state(repo_path: &str) -> Result<LocalState, Error> {
     let repo = git::Repository::new(repo_path)?;
-    // TODO(finto): This should be the default branch of the project, possibly.
-    let browser = Browser::new(&repo, git::Branch::local("master"))?;
+
+    let repoo = git2::Repository::open(repo_path)?;
+    let first_branch = repoo
+        .branches(Some(git2::BranchType::Local))?
+        .filter_map(|branch_result| {
+            let branch = match branch_result {
+                Ok(s) => s,
+                Err(_) => return None,
+            };
+
+            match branch.0.name() {
+                Ok(Some(n)) => Some(n.to_owned()),
+                Ok(None) | Err(_) => None,
+            }
+        })
+        .min()
+        .expect("Could not find any branches.");
```

Comment:
Should we panic for this case?

Reply:
This does indeed seem a bit dangerous -- a freshly initialised repo (`git init`) will return an empty `Branches` iterator, so it's not really a should-never-happen situation.

How about this?:

```rust
 let default_branch = repo
        .branches(Some(git2::BranchType::Local))?
        .filter_map(|branch_result| {
            let (branch, _) = branch_result.ok()?;
            let name = branch.name().ok()?;
            name.map(String::from)
        })
        .min_by(|x, y| match (x.as_ref(), y.as_ref()) {
            ("master", _) => Ordering::Less,
            (_, "master") => Ordering::Greater,
            _ => x.cmp(y)
        }).
        .ok_or_else(Error::UninitialisedRepository(repo_path))?;
```

Now, "master" is returned if it exists, and otherwise the lexicographically smallest branch. Or none at all, which should be a non-fatal error.